frontend/app/model/FeatureChoice.scala

```diff
@@ -0,0 +1,32 @@
+package model
+
+trait FeatureChoice {
+  val id: String
+  val label: String
+}
+
+object FeatureChoice {
+  val all: Set[FeatureChoice] = Set(
+    Events,
+    Books
+  )
+
+  val codes = all.map(_.id)
+
+  def fromString(id: String): Set[FeatureChoice] = id match {
+    case Books.id => Set(Books)
+    case Events.id => Set(Events)
+    case _ =>
+      if (id == Events.id + Books.id || id == Books.id + Events.id) Set(Books, Events)
```

Comment:
I think this parsing code would probably be better written to just split a comma-separated string, then individually map each element to a `FeatureChoice` in a Set